### DISCUSSION TITLE
CH03_07 AWS CLI 도구 설치 및 사용 강의 관련 질문 입니다.

### DISCUSSION THREAD

**Discussion by doctrin** [tedilabs/fastcampus-devops#22] [Q&A] [answered]
_**1.**_
로컬PC에 AWS CLI 서비스를 설치 하려고 보니
강의에서 지정한 특정 버전(2.2.16)은 정상적인 zip 파일이 다운로드 되는 것 같지를 않습니다 .
unzip 실행 시에 아래와 같은 내용의 에러가 발생합니다. 

![패스트캠퍼스_문의사항_ch03_07](https://user-images.githubusercontent.com/2967764/138557938-23247697-527c-4031-a550-77bd4a204bdc.png)

_**2.**_
위와 같은 에러로 인해 
curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
명령어로 최신의 aws cli 파일을 다운로드 받아서 unzip 까지 정상적으로 되었습니다. 

그런데 
aws sts get-caller-identity 명령어를 실행하면 또 에러가 발생합니다. 

![패스트캠퍼스_문의사항_ch03_07_02](https://user-images.githubusercontent.com/2967764/138558029-9f6f1e40-d0a9-424e-ae42-987e8c520464.png)

이 에러의 발생원인은 강의에서 지정한 2.2.16 버전을 설치하지 않고
최신 버전을 설치해서 발생한 문제일까요?

_**3.**_
aws ec2 describe-key-pairs 명령어를 실행 해봐도

역시 
Could not connect to the endpoint URL: "https://ec2.ap-northeast2.amazonaws.com/" 
라는 에러메시지가 발생합니다. 

_**4.**_
현재 저는 window10 환경에서 WSL을 사용하고 있습니다.
강의 영상에서 aws cli 설치 위치가 

~/.aws 라고 나오는건  '맥 OS' 환경이어서 그런것일까요?

왜냐면 저는 CH03_08번 강좌 보면서 진행을 하니까
그냥 ~/aws 폴더에 설치 되더군요.  **( not ~/.aws )**

혹시나 이것 때문에 문제가 발생된것은 아닐까 
적어봤습니다 .

해당 현상 확인 부탁드리겠습니다. 

답변 기다리고 있겠습니다. 
감사합니다 :D

**Comment by doctrin** [comment, marked as answer]
위의 질문 작성한 당사자 인데,

CH03_08 강의를 다시 보니 
aws에 다시 ec2 인스턴스를 하나 만들고 거기서 해보니 되는군요.

CH03_06 에서 만들었던 인스턴스를 해당 강좌에서
종료를 해서 없애버려서 그런 줄로 알고 계속 강좌를 봤는데

CH03_07 에서는 로컬 맥OS 에서 만들고
CH03_08 에서는 갑자기 aws ec2 인스턴스에서 진행하는게 해당 강좌에서 의도한게 맞나요?

확인 바랍니다.

**Reply by posquit0** [reply]
안녕하세요,

데브옵스 초격차 강의는 macOS, Ubuntu Linux, Windows WSL 사용자를 대상으로 진행됩니다.
현업에서 Windows를 사용하는 경우는 극히 드물지만, 패스트캠퍼스 측의 요청에 따라 Windows에 Ubuntu Linux 환경을 만들 수 있는 Windows WSL 기능을 소개하여 윈도우 사용자분들도 강의를 따라올 수 있도록 구성해보았습니다.

다만, 제가 Windows WSL 사용자가 아니다보니 Windows WSL Ubuntu와 Ubuntu 두 환경이 동일하다고 가정하고 강의를 진행중에 있습니다.
하여 강의는 macOS와 Ubuntu 두 가지에 대해 커버되며, macOS 사용자가 아닌 분들은 강의 제목에 (Ubuntu) 가 포함된 것을 수강하시면 됩니다.

CH03_07은 macOS 대상 강의이며, CH03_08은 macOS를 사용하지 않는 분들을 위한 강의입니다.

혹시나 Windows WSL 환경에서 계속하여 이슈가 발생한다면 AWS EC2의 Free-tier를 이용하여 Ubuntu 머신을 생성 후에 진행하시는 것을 추천드립니다.

**Reply by doctrin** [reply]
답변 감사합니다. 

그런데
제가 질문 드린 포인트를 다시 구체적으로 설명 드리자면 

>>> **AWS CLI 서비스**를 local WLS 기반으로 돌리고 있는 Ubuntu에 설치하는게 아니고,
        _**AWS 서비스에 생성한 EC2 인스턴스에 설치하는게 맞는지?**_ 를 질문 드리고 있는 것입니다. 

        **(즉,  local (WSL 기반의 Ubuntu) 에다가???   or    AWS EC2 instance 에다가???)**

만약,
위의 질문에 대한 답이 아마존 EC2 인스턴스에 설치하는게 답이라고 한다면

결국 local WSL-Ubuntu 의 역할은 아마존 EC2 인스턴스 서비스에 
접속하기 위한 일종의 게이트웨이 의 역할만 있는 것처럼 생각되어 집니다. 
(until CH03_08 까지의 강좌의 내용에 의하면)

그렇다면 굳이 WSL 서비스를 활용해서 로컬에서 Ubuntu를 깔아서
접속할 필연성 이라는것은 없는 것처럼 역시 생각 되어 지고요.
_(putty 로도 aws 접속은 되기에 ( https://docs.aws.amazon.com/ko_kr/AWSEC2/latest/UserGuide/putty.html )_

물론, 
Virtual Box 등과 같은 샌드박스 보다는 간편해서 
쓰기 편한건 있습니다.  :D

이 부분에 대한 확인 요청 드리도록 하겠습니다. 

읽어주셔서 감사합니다.

**Reply by posquit0** [reply]
안녕하세요,
제가 다시 Windows 10 환경에 WSL2 Ubuntu 20.04 LTS 구성하여 확인 후 답변드립니다.

1. 발표자료에 나온 명령어를 순서대로 입력하여 실행해보았으나, 특정 버전 aws-cli zip 파일도 잘 압축이 풀리며 설치도 잘 진행됩니다. 네트워크 이슈로 파일이 깨진게 아닐까 의심됩니다. 다시 다운로드 받아보시길 바랍니다.
2. 다시 한 번 오류 메시지를 확인해보니 URL 상에 `ap-northeast2`를 보고 잘못된 것을 찾았습니다. `~/.aws/config` 상에 아래처럼 `region` 값을 설정바랍니다. `ap-northeast2`가 아니라 `ap-northeast-2` 입니다.
```
[default]
region=ap-northeast-2
``` 
3. 2와 답변 동일합니다.
4. `aws` 디렉토리는 `aws-cli` zip 파일을 다운로드받아 압출을 풀면 나오는 디렉토리입니다. `~/.aws/config` 파일은 `aws-cli`가 참조하는 사용자 설정파일 경로입니다. 설치와 무관하며 직접 디렉토리 생성하셔서 진행하셔야 합니다.

또 궁금한 점 있으시면 남겨주세요.
감사합니다. :)

**Reply by doctrin** [reply]
답변 감사합니다. 

1. 답변 피드백     
 : 해당 파일을 다시 파일 다운로드 해보도록 하겠습니다. 

2 & 3. 답변 피드백 
: 오타 때문이었군요. 확인 감사합니다. 

4. 답변 피드백 
: 네. 이해 했습니다. 



그런데, 
위에서 부터 계속 말씀 드렸었고, 
좀 헷갈리는 부분이 

AWS CLI 서비스의 설치 머신인데요.

-------------------------------------------------------------------------------------------

**AWS CLI 서비스를**

**A.**
**-> local의  WLS 기반으로 돌리고 있는 Ubuntu에 설치 ?**

**B.**
**-> AWS 서비스에 생성한 EC2 인스턴스(이것 역시 WSL2 Ubuntu 20.04 LTS)에 설치하는게 맞는지?** 

**를 질문 드렸습니다.**

-------------------------------------------------------------------------------------------



답변해주시는 내용으로 보건데,

**AWS CLI 서비스를
Local 의 WSL2 Ubuntu 20.04 LTS (not AWS EC2 Instance) 
에다가 설치** 하는것으로 
유추 되는데요.


답변 해주시느라 많이 힘드셨을텐데
제 질문의 답이 

**_AWS CLI 서비스를 Local 의 WSL2 Ubuntu 20.04 LTS (not AWS EC2 Instance) 
에다가 설치 하는것이 맞다_**  

이면, 간단하게 **'ㅇㅋ'** 만 달아주셔도 좋겠습니다. 


감사합니다~ :D

**Reply by posquit0** [reply]
실습 환경은 수강자 분께서 편하신 환경을 선택해 진행하시면 될 것 같습니다.
Windows WSL 환경은 윈도우 사용자에게 편하지만 일반적인 리눅스 환경이 아니기 때문에 예기치 못한 오류가 발생할 수 있습니다.
강의에서는 패스트캠퍼스의 요구에 따라 여러 환경에 대해 가이드를 드릴 수 밖에 없는 점 양해 부탁드립니다.
